### ISSUE TITLE
when using test module option with value None the get_opt() returns string 'None' instead of None

### ISSUE THREAD

**Issue by jtluka** [LNST-project/lnst#158]
Actually this breaks phase1/phase2 regression tests if user removes nperf_confidence alias. The Netperf code compares with None and this would fail no matter what the value of nperf_confidence is.

**Comment by olichtne** [comment]
Turns out this needs a more complex rework of how test module options work. Currently the options are transfered in a dictionary that looks like this:
```python
{'bind': [{'value': '192.168.100.1'}],
 'confidence': [{'value': 'None'}],
 'cpu_util': [{'value': 'None'}],
 'debug': [{'value': '0'}],
 'duration': [{'value': '60'}],
 'max_deviation': [{'value': '20%'}],
 'netperf_opts': [{'value': ''}],
 'netperf_server': [{'value': '192.168.100.2'}],
 'num_parallel': [{'value': '1'}],
 'role': [{'value': 'client'}],
 'runs': [{'value': '5'}],
 'testname': [{'value': 'TCP_STREAM'}]}
```
So... each option is a list of {'value': string_value} dictionaries. The reason for this is the original way of how we worked with module options from XML tasks. The Python Tasks use a backwards compatible approach and translate a normal python dictionary of values into the old format, this results in None "values" getting traslated into "None" strings.

The natural solution is to just use the Python dictionary and transport it to the Slave without change, which is not possible from the still supported XML tasks.

Splitting the module option handling into two separate code paths would be complicated and hard to support. Because of this I'm removing this issue from the "Release 12" milestone and I'll instead rework this once we switch to purely Python recipes.
